uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/LoginIT.java

```diff
@@ -269,6 +269,13 @@ public void testPasscodeRedirect() {
         attemptLogin(testAccounts.getUserName(), testAccounts.getPassword());
 
         assertThat(webDriver.findElement(By.cssSelector("h1")).getText(), Matchers.containsString("Temporary Authentication Code"));
+
+        // Verify that the CopyToClipboard function can be executed
```

Comment:
It seems to me like this test is mixing 2 different layers of abstraction, by testing the javascript method, and then testing the UI.

This is an integration test, it seems to me like we could just test that the functionality is working, when we click on the button.

Reply:
By testing both the button's existence and it is clickable, and also the Java function is loaded and can be executed, we at least test both components involved in the action.  I do understand that we miss the part verifying that the function is being executed when you click the button.